lib/IDE/CodeCompletion.cpp

```diff
@@ -373,13 +373,18 @@ CodeCompletionString *CodeCompletionString::create(llvm::BumpPtrAllocator &Alloc
 }
 
 void CodeCompletionString::print(raw_ostream &OS) const {
+
   unsigned PrevNestingLevel = 0;
+  SmallVector<StringRef, 3> closeTags;
+
   auto chunks = getChunks();
   for (auto I = chunks.begin(), E = chunks.end(); I != E; ++I) {
     bool AnnotatedTextChunk = false;
 
-    if (I->getNestingLevel() < PrevNestingLevel) {
-      OS << "#}";
+    while (I->endsPreviousNestedGroup(PrevNestingLevel)) {
+      OS << closeTags.back();
+      closeTags.pop_back();
+      --PrevNestingLevel;
```

Comment:
Why do we modify this here?  I would have expected this to be unmodified until the *next* loop iteration.

Reply:
Let's say we have:
```
0 (Text "out")
1 | (G1Begin)
3 | (Text "g1")
2 | | (G2Begin)
3 | | (Text "g2")
4 | (G3Begin)
5 | (Text "g3")
```
and the group tag is always `{...}`
After printing item `3`.  the output so far is `out{g1{g2`, `PrevNestingLevel` is 2, `closeTags` are `["}", "}"]`
Since item `4`(G3Begin) is `endsPreviousNestedGroup()` for both `2` and `1`, we want to print `}}` here, so that the final output becomes `out{g1{g2}}{g3}`.

I don't think it's possible to do this without modifying `PrevNestingLevel`. WDYT?
Alternatively, since `closeTags.size()` should always equal to `PrevNestingLevel`, we can eliminate `PrevNestingLevel` with appropriate `assert`s.

Reply:
Okay, that makes some sense. I *think* it might be easier to understand without PrevNestingLevel, but we'd need to change it and see I guess.  No need to change for this PR, thanks for explaining.